woocommerce/woocommerce-android
# Issue/4413 unsupported country

Parent issue #4413 

This PR
- Refactors OnboardingState enum to a sealed class 
- Adds support for showing display name of Store's country on the Ui when Unsupported country state shown
- Implements "learn more" and "contact support" on click listeners - the contact support listener is still not implemented in the fragment since I didn't want to introduce a new origin before https://github.com/woocommerce/woocommerce-android/issues/4421 is implemented

To test:
- This flow is not reproducible yet, since the In Person Payments UI is disabled on stores without WCPay and in non-us countries. You can return COUNTRY_NOT_SUPPORTED from[ this method](https://github.com/woocommerce/woocommerce-android/blob/18c64f1610e9a24ecf4a0b409beaadbcae5ca4e3/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingChecker.kt#L29) to test this flow.
1. Open App settings
2. Tap on "InPerson Payments"
2. Notice the header contains name of store's country

<img width="250px" alt="Screenshot 2021-07-27 at 7 16 36" src="https://user-images.githubusercontent.com/2261188/127099640-c54e434e-88f1-48a3-91f6-1bf188e016aa.png">



Update release notes:
